open-source-hosts-plugin/test/workspace.test.ts

```diff
@@ -22,23 +23,71 @@ describe('workspace', () => {
   });
 
   describe('prepareWorkspace', () => {
-    it('should call copyFile and mkdir once separately', async () => {
+    it('should copy scripts to workspace', async () => {
       const mockCopy = jest.fn();
       jest.spyOn(fs, 'copy').mockImplementation(mockCopy);
+
       await prepareWorkspace('testPluginDir', 'testWorkSpaceDir');
 
-      expect(fs.promises.copyFile).toHaveBeenCalledTimes(1);
-      expect(fs.promises.mkdir).toHaveBeenCalledTimes(1);
-      expect(fs.copy).toHaveBeenCalledTimes(1);
+      const fromPath = path.join('testPluginDir', 'scripts', 'AwsTools');
+      const toPath = path.join('testWorkSpaceDir', 'src', 'scenes', 'AwsTools');
+
+      expect(mockCopy).toHaveBeenCalledWith(
+        fromPath,
+        toPath,
+        expect.anything()
```

Comment:
`const mockCopy = jest.fn()` is just our way of saying "use a dummy function in place of the real method". We don't have to explicitly tell it how many arguments to accept since our dummy function is a no-op and since JavaScript doesn't do type checking of function signatures. When our dummy function gets called by the code under test, it actually gets called with 3 arguments, so our "expects()" assertion is confirming that the first two arguments it receives are what we expect. The outcome we're testing for is not affected by the third argument which is why we're accepting any value as the third arg.

Reply:
I see. Thank you for the explanation.